agent_app_sample_code/02_data_pipeline.py

```diff
@@ -0,0 +1,421 @@
+            }
+        elif file_extension == "docx":
+            with tempfile.NamedTemporaryFile(delete=True) as temp_file:
+                temp_file.write(raw_doc_contents_bytes)
+                temp_file_path = temp_file.name
+                md = pypandoc.convert_file(temp_file_path, "markdown", format="docx")
+
+                parsed_document = {
+                    "doc_content": md.strip(),
+                    "parser_status": "SUCCESS",
+                }
+        else:
+            raise Exception(f"No supported parser for {doc_path}")
+
+        # Extract the required doc_uri
+        # convert from `dbfs:/Volumes/catalog/schema/pdf_docs/filename.pdf` to `Volumes/catalog/schema/pdf_docs/filename.pdf`
+        modified_path = doc_path[5:]
+        parsed_document["doc_uri"] = modified_path
+
+        # Sample metadata extraction logic
+        if "test" in parsed_document["doc_content"]:
+            parsed_document["example_metadata"] = "test"
+        else:
+            parsed_document["example_metadata"] = "not test"
+
+        # Add the modified time
+        parsed_document["last_modified"] = modification_time
+
+        return parsed_document
+
+    except Exception as e:
+        status = f"An error occurred: {e}\n{traceback.format_exc()}"
+        warnings.warn(status)
+        return {
+            "doc_content": "",
+            "parser_status": f"ERROR: {status}",
+        }
+
+# COMMAND ----------
+
+# MAGIC %run ./utils/typed_dicts_to_spark_schema
+
+# COMMAND ----------
+
+# MAGIC %run ./utils/load_uc_volume_to_delta_table
+
+# COMMAND ----------
+
+load_uc_volume_to_delta_table(
+    source_path=SOURCE_UC_VOLUME,
+    dest_table_name=DOCS_DELTA_TABLE,
+    # Modify this function to change the parser, extract additional metadata, etc
+    parse_file_udf=file_parser,
+    # The schema of the resulting Delta Table will follow the schema defined in ParserReturnValue
+    spark_dataframe_schema=typed_dicts_to_spark_schema(ParserReturnValue),
+)
+
+print(DOCS_DELTA_TABLE)
+display(spark.table(DOCS_DELTA_TABLE))
+
+# Log the resulting table to MLflow
+mlflow.log_input(
+    mlflow.data.load_delta(
+        table_name=DOCS_DELTA_TABLE, name=DOCS_DELTA_TABLE.replace("`", "")
+    )
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Compute chunks of documents
+# MAGIC
+# MAGIC Now we will split our documents into smaller chunks so they can be indexed in our vector database.
+# MAGIC
+# MAGIC We've provided a utility `chunk_docs` to help with this. Alternatively, you could compute chunks in your own spark pipeline and pass it to the indexing step after chunking.
+# MAGIC
+
+# COMMAND ----------
+
+# MAGIC %run ./utils/chunk_docs
+
+# COMMAND ----------
+
+# MAGIC %run ./utils/get_recursive_character_text_splitter
+
+# COMMAND ----------
+
+# Configure the chunker
+chunk_fn = get_recursive_character_text_splitter(
+    model_serving_endpoint=EMBEDDING_MODEL_ENDPOINT,
+    chunk_size_tokens=384,  # 1024,
```

Comment:
? Why the comments here / should this be a config as well?

Reply:
we dont have a notion of a data pipeline config currently. we only have a config for the agent part. maybe we can tackle this in a followup?

Reply:
Yeah followup sounds good, maybe we can just drop the commented out stuff